File: docs/src/testers/sessions/lifecycle/create_session.ts
// @sniptest filename=create_session.ts
// @sniptest show=7-20
import { NotteClient, type SessionResponse } from 'notte-sdk';

const client = new NotteClient();
let status: SessionResponse | undefined;
let title: string | undefined;

// Recommended: Use .use() for automatic cleanup.
await client
  .Session({ viewport_width: 1920, viewport_height: 1080 })
  .use(async (session) => {
    console.log(`Session ${session.getId()} is active`);

    // Access the session’s Playwright page.
    status = await session.status();
    const page = await session.page();
    await page.goto('https://example.com');
    title = await page.title();
    console.log(`Page title: ${title}`);
  });
// Automatically stopped here.

export { status, title };

File: docs/src/testers/sessions/lifecycle/create_session.py
# @sniptest filename=create_session.py
# @sniptest show=5-12
from notte_sdk import NotteClient

client = NotteClient()

# Recommended: Use context manager for automatic cleanup
with client.Session(viewport_width=1920, viewport_height=1080) as session:
    print(f"Session {session.session_id} is active")

    # Access Playwright page
    page = session.page
    page.goto("https://example.com")
    print(f"Page title: {page.title()}")

    # Values retained for the external test runner, outside the displayed range.
    status = session.status()
    title = page.title()
    viewport = page.evaluate("({width: window.innerWidth, height: window.innerHeight})")
